### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by adi2011**
Recurrence class implementation, signature verification, Offer_checks,fetchinvoice api..

### CODE REVIEW

File: javascript/index.js [hunk 1]
```diff
@@ -29,7 +166,8 @@ function taggedHash(tag, msg) {
     const tagHash = hash(tag);
     return hash(concat([tagHash, tagHash, Buffer.from(msg,'hex')]));
 }
-
+console.log(taggedHash( Buffer.from('lightningoffersignature'),
+                '28522b52ac39fa518ce3a5b3e4a9a96372487e78ba5eb1540ec4d9f02ca82718').toString('hex'))
 function convert (data, inBits, outBits) {
```
Comment (adi2011) [inline javascript/index.js:182]: Will remove these stupid debug lines after invoice_request function..

**Review by adi2011 — COMMENTED** [review]
``` 2d43373```
Not getting good vibes from this commit, sorry in advance.

File: javascript/index.js [hunk 2]
```diff
@@ -353,8 +380,25 @@ function decode(paymentReq){
         if(tagCode<240||tagCode>1000)
             tags.push(Buffer.concat([Buffer.from(tlvs.slice(0,2)),tlvs[2]]).toString('hex'));
     }
+    final['offer_id']=merkle_calc(tags);
+    if(!'discription' in fin_content ||!'node_id' in fin_content){
```
Comment (rustyrussell) [inline javascript/index.js]: s/discription/description/ everywhere...

**Review by rustyrussell — APPROVED** [review]
Only real feedback is the obvious typo!  Please fix and I'll remove the merge commit and apply to master.

File: javascript/index.js (hunk 2 above)
Comment (adi2011) [reply javascript/index.js]: Done..

**Review by rustyrussell — APPROVED** [review]

**Pull request merged**